src/ape/api/providers.py

```diff
@@ -707,47 +706,42 @@ def get_transactions_by_block(self, block_id: BlockID) -> Iterator:
 
     def get_contract_logs(
         self,
-        address: Union[AddressType, List[AddressType]],
-        abi: Union[List[EventABI], EventABI],
-        start_block: Optional[int] = None,
-        stop_block: Optional[int] = None,
+        log_filter: LogFilter,
         block_page_size: Optional[int] = None,
-        event_parameters: Optional[Dict] = None,
     ) -> Iterator[ContractLog]:
```

Comment:
could we also add reasonable parallelism to `get_contract_logs`? it's what people do anyway, might as well be the default. something like `concurrent.futures.ThreadPoolExecutor` with 4 workers would do.